contrib/linux/bash-completion/dosbox

```diff
@@ -0,0 +1,183 @@
+# bash_completion
+####
+_dosbox()
+{
+  COMPREPLY=()
+  local cur prev words cword
+  _init_completion -s || return
+  local IFS=$'|\t\n'
+  local words_filter="${words[*]:1:$(( cword - 1 ))}"
+
+  ### Main options, note the \ line cont.
+  # using nameref
+  # shellcheck disable=SC2034
+  local -a \
+       _main_opts=(
+                    -h --help -fullscreen -startmapper -noautoexec -securemode
+                    -scaler -forcescaler -lang -machine -socket -exit
+                  )\
+     _repeat_opts=(
+                    -c -conf
+                  )\
+  _exclusive_opts=(
+                    --version --printconf --editconf -resetconf -eraseconf
+                    -opencaptures -erasemapper -resetmapper
+                  )\
+  ####
+
+  # setup nameref to point to array
+  [[ ${prev:0:1} == - ]] && local -n _sub_opts=${prev//+(-)/_}
+
+  ### Arrays holding sub. options
+  # The name should be the trigger option with leading "-"s replaced with "_"
+  # for example, --machine => _machine
+  # latter parts of the script will reference these arrays via a "nameref"
+  # i.e. The last option dictates the available sub. options
+
+  #
+  # shellcheck disable=SC2034
+  local -a \
+         _machine=(
+                    hercules cga tandy pcjr ega vgaonly svga_s3 svga_et3000
+                    svga_et4000 svga_paradise vesa_nolfb vesa_oldvbe
+                  )\
+          _scaler=(
+                    none normal2x normal3x advmame2x advmame3x advinterp2x
+                    advinterp3x hq2x hq3x 2xsai super2xsai supereagle tv2x
+                    tv3x rgb2x rgb3x scan2x scan3x
+                  )\
+    _opencaptures=(
+                    "$BROWSER" firefox mpv pcmanfm pcmanfm-qt
+                  )\
+  # A note on -opencaptures, it launches the provided prog. and points it to
+  # the dir. containing captured media (screenshots, video, audio)
+  # TODO add more media browser programs
+
+  [[ $prev =~ ^-{1,2}opencaptures$ ]] &&
+    readarray -t -O 0 _opencaptures < <(
+      # do we have a custom wrapper script available?
+      [[ -x "$XDG_CONFIG_HOME/dosbox/opencaptures.sh" ]] &&
+        printf "%s\n" "$XDG_CONFIG_HOME/dosbox/opencaptures.sh"
+      # TODO provide example script
+      # may not be worth it..
+
+      # filter out unavailable media browsers
+      type -P "${_opencaptures[@]}"
+      # TODO document recommended "media browsers"
+    )
+  ####
+
+  ### Offer ability for the user to override the default .conf dir.
+  # shellcheck disable=SC2034
+  local _conf="${DOSBOX_STAGING_CONF_DIR:-$XDG_CONFIG_HOME/dosbox/}"
+  #TODO validate the dir exists
+  #TODO document DOSBOX_STAGING_CONF_DIR var.
+  ####
+
+  # forcescaler uses same sub. options as scaler, so setup a nameref to it
+  # using nameref _sub_opts
+  # shellcheck disable=SC2034
+  local -n _forcescaler=_scaler
+
+  # can we get hints from manpage?
+  man -w dosbox >/dev/null 2>&1 && local manpage_hints=true
```

Comment:
What does this do?

Reply:
it provides a hint when no completion is programmed

```
 -socket|-lang|-c)

```

providing manpage is available

```
dosbox -c <tab><tab>
» no completion support for -c yet
» hint from manpage:
» -c command
» Runs the specified command before running file.  Multiple commands can be specified. Each command should start
» with -c though. A command can be: an Internal Program, a DOS command or an executable on a mounted drive.
```

I populate the var. early so --MANPAGE is an option,
This launches the man, and on exit from it clears the entered --MANPAGE (well, inserts a space)

Reply:
Testing and working really nicely.

If you're in the middle of a command line and run out of <tab> suggestions (for example, `-socket needs a user value, the tab completion can't give that to you), then it recommends using --MANPAGE to learn more. So you can add `-M` <tab> pulls up the manpage, learn about your setting, quit, and then land back in the middle of your tab line and populate it with what you've learned.

Very nice :+1:. Resolving.